planning/behavior_velocity_planner/src/scene_module/crosswalk/manager.cpp

```diff
@@ -133,12 +125,55 @@ CrosswalkModuleManager::getModuleExpiredFunction(
   const autoware_auto_planning_msgs::msg::PathWithLaneId & path)
 {
   const auto rh = planner_data_->route_handler_;
-  const auto crosswalk_id_set =
-    getCrosswalkIdSetOnPath(path, rh->getLaneletMapPtr(), rh->getOverallGraphPtr());
+  const auto crosswalk_id_set = getCrosswalkIdSetOnPath(
+    planner_data_->current_pose.pose, path, rh->getLaneletMapPtr(), rh->getOverallGraphPtr());
 
   return [crosswalk_id_set](const std::shared_ptr<SceneModuleInterface> & scene_module) {
     return crosswalk_id_set.count(scene_module->getModuleId()) == 0;
   };
 }
 
+WalkwayModuleManager::WalkwayModuleManager(rclcpp::Node & node)
+: SceneModuleManagerInterface(node, getModuleName())
+{
+  const std::string ns(getModuleName());
+
+  // for walkway parameters
+  auto & wp = walkway_planner_param_;
+  wp.stop_margin = node.declare_parameter(ns + ".walkway.stop_margin", 1.0);
+  wp.stop_line_distance = node.declare_parameter(ns + ".walkway.stop_line_distance", 1.0);
+  wp.stop_duration_sec = node.declare_parameter(ns + ".walkway.stop_duration_sec", 1.0);
+  wp.external_input_timeout = node.declare_parameter(ns + ".walkway.external_input_timeout", 1.0);
+}
+
+void WalkwayModuleManager::launchNewModules(
+  const autoware_auto_planning_msgs::msg::PathWithLaneId & path)
+{
+  const auto rh = planner_data_->route_handler_;
+  for (const auto & crosswalk : getCrosswalksOnPath(
+         planner_data_->current_pose.pose, path, rh->getLaneletMapPtr(),
+         rh->getOverallGraphPtr())) {
+    const auto module_id = crosswalk.id();
+    if (
+      !isModuleRegistered(module_id) &&
+      crosswalk.attributeOr(lanelet::AttributeNamesString::Subtype, std::string("")) ==
+        lanelet::AttributeValueString::Walkway) {
+      registerModule(std::make_shared<WalkwayModule>(
+        module_id, crosswalk, walkway_planner_param_, logger_.get_child("walkway_module"), clock_));
```

Comment:
`generateUUID` is not needed?

Reply:
Thanks for the comment.
It is not necessary in the walkway module.
So, it is okay.

Reply:
OK, I understood. Whenever walkway is registered, crossway is also registered and already `generateUUID` with the same module_id.